test/integration/consul-container/test/envoy_extensions/wasm_test.go

```diff
@@ -0,0 +1,626 @@
+package envoy_extensions
+
+import (
+	"bufio"
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	libassert "github.com/hashicorp/consul/test/integration/consul-container/libs/assert"
+	libcluster "github.com/hashicorp/consul/test/integration/consul-container/libs/cluster"
+	libservice "github.com/hashicorp/consul/test/integration/consul-container/libs/service"
+	"github.com/hashicorp/consul/test/integration/consul-container/libs/topology"
+)
+
+// TestWASMRemote Summary
+// This test ensures that a WASM extension can be loaded from a remote file server then executed.
+// It uses the same basic WASM extension as the TestWASMLocal test which adds the header
+// "x-test:true" to the response.
+// This test configures a static server and proxy, a client proxy, as well as an Nginx file server to
+// serve the compiled wasm extension. The static proxy is configured to apply the wasm filter
+// and pointed at the remote file server. It should download the file and add it to the envoy filters.
```

Comment:
nit: The order of the last sentence isn't entirely correct. The filter is actually what triggers Envoy to download the Wasm file.